webclient/src/components/admin/categories/Banner.js

```diff
@@ -52,60 +56,119 @@ const BannerDetails = ({ banners, slug, setDetails }) => {
   );
 };
 
-const UpsertBanner = props => {
+const UpsertBanner = ({ slug, detailType, closeForm }) => {
   const [selectedFile, setSelectedFile] = useState(null);
-  const [
-    { response: imageURL, error: mutationError },
-    upsertData
-  ] = useMutation();
+  const [imageUrl, setImageUrl] = useState('');
+  const [uploadFile] = useMutation(postFile, {
+    onSuccess: data => {
+      setImageUrl(data?.image?.url);
+      queryCache.refetchQueries(QUERY_KEY_ADMIN_CATEGORY_DETAILS);
+    }
+  });
+  const [upsert] = useMutation(postUpdate, {
+    onSuccess: () => {
+      closeForm();
+      queryCache.refetchQueries(QUERY_KEY_ADMIN_CATEGORY_DETAILS);
+    }
+  });
 
-  const handleSelectFile = e => {
-    console.log(e.target.files[0]);
-    setSelectedFile(e.target.files[0]);
-    M.toast({
-      html: '<div>Remember to save!</div>',
-      classes: 'amber rounded center top'
+  function postFile(data) {
+    return client(API_FILE_UPLOAD, {
+      file: data,
+      showError: true
     });
-  };
+  }
 
-  const handleFileUpload = async e => {
-    e.preventDefault();
-    const data = new FormData();
-    data.append('file', selectedFile);
-
-    await upsertData({
-      method: 'post',
-      endpoint: `/images/upload`,
-      data
+  function postUpdate(data) {
+    client(API_ADMIN_CATEGORY + `/${slug}/${detailType}`, {
+      body: data,
+      showError: true,
+      showSuccess: true
     });
+  }
+
+  const checkMimeType = e => {
+    let files = e.target.files;

```

Comment:
hmm is it possible that there are multiple files

Reply:
Yeah, just in case you want to extend some time later. The code's there alr